### PR TITLE
Reload map on prefix+suffix change on category dataview

### PR DESCRIPTION
**Pull request by viddo**
Related to #1040, some things were missing in previous PRs

Also discovered that aggregation_column need to be translated to camelcased version when sent to Windshaft, added a todo for this for now so we can move fwd

Finally, moved attrs names definition here as well as some minimal validation (presence) of attrs when creating the dataview.

@alonsogarciapablo 
cc @xavijam

### CODE REVIEW

**Comment by alonsogarciapablo** [comment]
:+1:

File: src/dataviews/category-dataview-model.js [hunk 1]
```diff
@@ -44,6 +45,7 @@ module.exports = DataviewModelBase.extend({
     this._searchModel = new SearchModel();
 
     this.on('change:column change:aggregation change:aggregation_column', this._reloadMap, this);
+    this.on('change:prefix change:suffix', this._reloadMap, this);
```
Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: Do we need to reload the map when preffix or suffix changes?

File: src/dataviews/dataview-model-base.js [hunk 2]
```diff
@@ -178,4 +182,14 @@ module.exports = Model.extend({
     this.trigger('destroy', this);
     this.stopListening();
   }
-});
+},
+
+  // Class props
+  {
+    ATTRS_NAMES: [
+      'sync_on_data_change',
```
Comment (xavijam) [inline src/dataviews/dataview-model-base.js]: :+1: 

cc @javierarce

File: src/dataviews/category-dataview-model.js [hunk 3]
```diff
@@ -248,11 +250,25 @@ module.exports = DataviewModelBase.extend({
       options: {
         column: this.get('column'),
         aggregation: this.get('aggregation'),
-        aggregation_column: this.get('aggregation_column'),
-        suffix: this.get('suffix'),
-        prefix: this.get('prefix')
+
+        // TODO server-side is using camelCased attr name, update once fixed
+        aggregationColumn: this.get('aggregation_column'),
```
Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: Not needed to be changed in Maps API, cc @rochoa.

File: src/dataviews/category-dataview-model.js [hunk 4]
```diff
@@ -248,11 +250,25 @@ module.exports = DataviewModelBase.extend({
...
-
-});
+},
+
+  // Class props
+  {
+    ATTRS_NAMES: DataviewModelBase.ATTRS_NAMES.concat([
+      'column',
+      'aggregation',
+      'aggregation_column',
```
Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: cc @javierarce

File: src/dataviews/histogram-dataview-model.js [hunk 5]
```diff
@@ -100,4 +109,16 @@ module.exports = DataviewModelBase.extend({
...
-});
+},
+
+  // Class props
+  {
+    ATTRS_NAMES: DataviewModelBase.ATTRS_NAMES.concat([
+      'column',
+      'column_type',
+      'bins',
+      'start',
```
Comment (xavijam) [inline src/dataviews/histogram-dataview-model.js]: I feel start and end are attributes for time-series, aren't they @javierarce?

File: src/dataviews/category-dataview-model.js (hunk 1 above)
Comment (viddo) [inline src/dataviews/category-dataview-model.js]: They're included in the data sent to Windshaft, so judging from that I'd say yes, but @rochoa or @alonsogarciapablo can perhaps confirm?

File: src/dataviews/histogram-dataview-model.js (hunk 5 above)
Comment (viddo) [inline src/dataviews/histogram-dataview-model.js]: Yes, only `column` and `bins` are required for a normal histogram, `column_type`, `start` and `end` is required (set to `date` from DI) for the time-series use-case.

File: src/dataviews/category-dataview-model.js (hunk 1 above)
Comment (viddo) [inline src/dataviews/category-dataview-model.js]: After talking with @javierarce I understand that these are not needed for maps API or dataviews, so extracted them up to DI.

**Pull request merged**